src/urllib3/connectionpool.py

```diff
@@ -7,6 +7,11 @@
 from socket import error as SocketError, timeout as SocketTimeout
 import socket
 
+try:
+    from http.client import RemoteDisconnected
+except ImportError:
+    from httplib import BadStatusLine as RemoteDisconnected
```

Comment:
This line will catch other ```BadStatusLine``` exceptions that are not remote disconnection and might resend the request when inappropriate.

Reply:
@hodbn in https://hg.python.org/cpython/rev/eba80326ba53 they noted this for exception exception:: RemoteDisconnected  ->  "  Previously, :exc:`BadStatusLine`\ ``('')``  
 was raised.  "

Reply:
Only until [that commit](https://github.com/python/cpython/commit/ee78ba2c819b0cd7671a575e584d8fe1b7adb3e4) 😞

Reply:
Ok, If I understand correctly, I have to distinguish if BadStatusLine("No status line received - the server has closed the connection") or BadStatusLine(something) for py2. Do you have some recommendation how I should do it ?

Reply:
I can't think of a way that does not involve handling all three version ranges differently and peeking inside ```BadStatusLine``` for the newly introduced message 😞

Reply:
Quick question: since the old message contained the actual line could we see if `BadStatusLine.args[0] == ""`?

Reply:
Yeah, that should be ok with ```< 2.7.16```.

Reply:
It seems silly but could we do `BadStatusLine.args[0] in ("", "No status line received - the server has closed the connection")`? The worst that'd happen is we'd retry on a "legit" bad status error because some host decided to add that as its actual HTTP status.

Reply:
> It seems silly but could we do `BadStatusLine.args[0] in ("", "No status line received - the server has closed the connection")`? The worst that'd happen is we'd retry on a "legit" bad status error because some host decided to add that as its actual HTTP status.

Hmm, but it looks like everytime when BadStatusLine is raised urllib should retry request. And my patch is doing this ...   https://github.com/urllib3/urllib3/blob/f0c43419e45db17a3c5f287be80aa0d14e2f58c6/test/with_dummyserver/test_socketlevel.py#L635